src/err/mod.rs

```diff
@@ -225,18 +225,30 @@ impl PyErr {
         PyErr::from_state(state)
     }
 
+    /// Deprecated form of [`PyErr::get_type_bound`].
+    #[cfg_attr(
+        not(feature = "gil-refs"),
+        deprecated(
+            since = "0.21.0",
+            note = "`PyErr::get_type` will be replaced by `PyErr::get_type_bound` in a future PyO3 version"
+        )
+    )]
+    pub fn get_type<'py>(&'py self, py: Python<'py>) -> &'py PyType {
+        self.get_type_bound(py).into_gil_ref()
+    }
+
     /// Returns the type of this exception.
     ///
     /// # Examples
     /// ```rust
-    /// use pyo3::{exceptions::PyTypeError, types::PyType, PyErr, Python};
+    /// use pyo3::{exceptions::PyTypeError, types::PyType, PyErr, Python, prelude::PyAnyMethods};
     ///
     /// Python::with_gil(|py| {
     ///     let err: PyErr = PyTypeError::new_err(("some type error",));
-    ///     assert!(err.get_type(py).is(PyType::new::<PyTypeError>(py)));
+    ///     assert!(err.get_type_bound(py).is(PyType::new::<PyTypeError>(py)));
     /// });
     /// ```
-    pub fn get_type<'py>(&'py self, py: Python<'py>) -> &'py PyType {
+    pub fn get_type_bound<'py>(&'py self, py: Python<'py>) -> Bound<'py, PyType> {
```

Comment:
Here we can relax the lifetime bound on `self` as the returned `Bound` doesn't need to connect to it:

```suggestion
    pub fn get_type_bound<'py>(&self, py: Python<'py>) -> Bound<'py, PyType> {
```

(It was necessary to have that in the GIL Refs API because it was returning a GIL Ref borrowed from this `PyErr`.)